### PR TITLE
Use api model for grype groups and use provider for toggle feed enabled endpoint

### PR DESCRIPTION
**Pull request by zburstein**
Abandons the use of the legacy FeedGroupMetadata class for persisting groups in grype. Also uses the provider to toggle the feed

### CODE REVIEW

File: anchore_engine/services/policy_engine/api/controllers/feeds.py [hunk 1]
```diff
@@ -153,23 +130,23 @@ def sync_feeds(sync=True, force_flush=False):
...
+        return jsonify(feed.to_json()), 200
 
-        return jsonify(updated), 200
-    except AnchoreApiError:
-        session.rollback()
-        raise
+    except InvalidFeed:
+        raise BadRequest(
+            message="Feed not enabled on configured vulnerability provider",
+            detail={"feed": feed, "configured_provider": provider.__config__name__},
```
Comment (nightfurys) [inline anchore_engine/services/policy_engine/api/controllers/feeds.py:146]: use `provider.get_config_name()` The error message is a bit confusing. It should be something to the effet of "Feed not supported"

File: anchore_engine/services/policy_engine/engine/feeds/feeds.py [hunk 2]
```diff
@@ -984,45 +987,60 @@ def _set_group_counts(self, db, source_counts: List[RecordSource]) -> None:
...
     def _update_group_timestamps(self, db):
         """
         Updates the tiemstamps
         """
-        groups = [FeedGroupMetadata(**group) for group in self.grypedb_meta.groups]
 
-        for group in groups:
-            group.last_sync = self.grypedb_meta.synced_at
-            group.last_update = self.grypedb_meta.last_updated
+        if self.grypedb_meta.groups and isinstance(self.grypedb_meta.groups, List):
```
Comment (nightfurys) [inline anchore_engine/services/policy_engine/engine/feeds/feeds.py:1022]: instance type check should be for `list`, not `List`

File: anchore_engine/services/policy_engine/engine/feeds/feeds.py [hunk 3]
```diff
@@ -1255,6 +1259,21 @@ def _sync_grype_groups(
...
+
+            results.append(
+                GroupSyncResult(
+                    group=source.group,
+                    updated_record_count=updated_count,
+                    total_time_seconds=total_time_seconds,
+                    status="success",
+                )
+            )
+
```
Comment (nightfurys) [inline anchore_engine/services/policy_engine/engine/feeds/feeds.py:1276]: If this block is executed in the happy path, it should be moved into try possibly before the db.commit()

File: anchore_engine/services/policy_engine/engine/vulns/providers.py [hunk 4]
```diff
@@ -67,13 +64,17 @@
...
 from anchore_engine.subsys import logger, metrics
-from anchore_engine.utils import rfc3339str_to_datetime, timer
+from anchore_engine.utils import timer
 
 from .dedup import get_image_vulnerabilities_deduper, transfer_vulnerability_timestamps
 from .scanners import GrypeScanner, LegacyScanner
 from .stores import ImageVulnerabilitiesStore, Status
 
 
+class InvalidFeed(Exception):
```
Comment (nightfurys) [inline anchore_engine/services/policy_engine/engine/vulns/providers.py:74]: (nit) Not necessary for now but these exceptions should be defined in a common place so others can import them without importing the entire provider module. We should also have a base class with common behaviour and extend it. Otheriwse there's little point in raising custom exception

File: anchore_engine/services/policy_engine/engine/vulns/providers.py [hunk 5]
```diff
@@ -180,6 +223,30 @@ def update_feed_group_counts(self) -> None:
         """
         ...
 
+    def toggle_feed_enabled(
```
Comment (nightfurys) [inline anchore_engine/services/policy_engine/engine/vulns/providers.py:226]: (nit) This is not the most descriptive name. I know it's copied over. How about `update_feed_status` or something like it

**Review by nightfurys — COMMENTED** [review]
Nice refactoring all around, feed responses from providers look much better!

**Review by Vijay-P — APPROVED** [review]
Nice work. I agree with Swathi's comments.

File: anchore_engine/services/policy_engine/engine/vulns/providers.py (hunk 4 above)
Comment (zburstein) [reply anchore_engine/services/policy_engine/engine/vulns/providers.py:74]: Agreed. One benefit to this right now is that I can distinguish it from a ValueError in the controller and raise a more specific BadRequest error

**Review by nightfurys — APPROVED** [review]
lgtm

**Pull request merged**